elytron/src/main/java/org/wildfly/extension/elytron/AdvancedModifiableKeyStoreDecorator.java

```diff
@@ -680,6 +721,245 @@ protected void executeRuntimeStep(final OperationContext context, final ModelNod
+            String reason = REASON.resolveModelAttribute(context, operation).asStringOrNull();
+            String certificateAuthorityAccountName = CERTIFICATE_AUTHORITY_ACCOUNT.resolveModelAttribute(context, operation).asString();
+            boolean staging = STAGING.resolveModelAttribute(context, operation).asBoolean();
+
+            try {
+                if (! keyStore.containsAlias(alias)) {
+                    throw ROOT_LOGGER.keyStoreAliasDoesNotExist(alias);
+                }
+                X509Certificate certificateToRevoke = (X509Certificate) keyStore.getCertificate(alias);
+                if (certificateToRevoke == null) {
+                    throw ROOT_LOGGER.unableToObtainCertificate(alias);
+                }
+                final AcmeAccount acmeAccount = getAcmeAccount(context, certificateAuthorityAccountName);
+                if (reason != null) {
+                    acmeClient.revokeCertificate(acmeAccount, staging, certificateToRevoke, getCRLReason(reason));
+                } else {
+                    acmeClient.revokeCertificate(acmeAccount, staging, certificateToRevoke);
+                }
+                keyStore.deleteEntry(alias);
+                ((KeyStoreService) keyStoreService).save();
+            } catch (IllegalArgumentException e) {
+                throw new OperationFailedException(e);
+            } catch (KeyStoreException | AcmeException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        static CRLReason getCRLReason(String reason) throws OperationFailedException {
+            switch (reason.toUpperCase(Locale.ENGLISH)) {
+                case UNSPECIFIED:
+                    return CRLReason.UNSPECIFIED;
+                case KEY_COMPROMISE:
+                    return CRLReason.KEY_COMPROMISE;
+                case CA_COMPROMISE:
+                    return CRLReason.CA_COMPROMISE;
+                case AFFILIATION_CHANGED:
+                    return CRLReason.AFFILIATION_CHANGED;
+                case SUPERSEDED:
+                    return CRLReason.SUPERSEDED;
+                case CESSATION_OF_OPERATION:
+                    return CRLReason.CESSATION_OF_OPERATION;
+                case CERTIFICATE_HOLD:
+                    return CRLReason.CERTIFICATE_HOLD;
+                case REMOVE_FROM_CRL:
+                    return CRLReason.REMOVE_FROM_CRL;
+                case PRIVILEGE_WITHDRAWN:
+                    return CRLReason.PRIVILEGE_WITHDRAWN;
+                case AA_COMPROMISE:
+                    return CRLReason.AA_COMPROMISE;
+                default:
+                    throw ROOT_LOGGER.invalidCertificateRevocationReason(reason);
+            }
+        }
+    }
+
+    static class ShouldRenewCertificateHandler extends ElytronRuntimeOnlyHandler {
+        static final SimpleAttributeDefinition EXPIRATION = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.EXPIRATION, ModelType.LONG, true)
```

Comment:
is it missing a measurement unit?

Reply:
"expiration" is the number of days to expiry to be checked.